File: Services/java-services/Core/CoreShared/src/main/java/ai/omnicure/core/shared/constant/PermissionConstants.java
package ai.omnicure.core.shared.constant;

public final class PermissionConstants {

    private PermissionConstants() {
        throw new UnsupportedOperationException("Constant class");
    }

    public static final class IAM {
        public static final String VIEW = "Permissions.IAM.View";
        public static final String CREATE = "Permissions.IAM.Create";
        public static final String EDIT = "Permissions.IAM.Edit";
        public static final String DELETE = "Permissions.IAM.Delete";
    }

    public static final class Patient {
        public static final String VIEW = "Permissions.Patient.View";
        public static final String CREATE = "Permissions.Patient.Create";
        public static final String EDIT = "Permissions.Patient.Edit";
        public static final String DELETE = "Permissions.Patient.Delete";
    }

    public static final class Appointment {
        public static final String VIEW = "Permissions.Appointment.View";
        public static final String CREATE = "Permissions.Appointment.Create";
        public static final String EDIT = "Permissions.Appointment.Edit";
        public static final String DELETE = "Permissions.Appointment.Delete";
    }

    public static final class Verification {
        public static final String VIEW = "Permissions.Verification.View";
        public static final String PROCESS = "Permissions.Verification.Process";
    }
}

File: Services/dotnet-services/Core/OmniCure.Core.Shared/Constants/PermissionConstants.cs
using OmniCure.Core.Shared.Helpers;

namespace OmniCure.Core.Shared.Constants;

public static class PermissionConstants
{
    public static class IAM
    {
        public const string View = "Permissions.IAM.View";
        public const string Create = "Permissions.IAM.Create";
        public const string Edit = "Permissions.IAM.Edit";
        public const string Delete = "Permissions.IAM.Delete";
    }

    public static class Patient
    {
        public const string View = "Permissions.Patient.View";
        public const string Create = "Permissions.Patient.Create";
        public const string Edit = "Permissions.Patient.Edit";
        public const string Delete = "Permissions.Patient.Delete";
    }

    public static class Appointment
    {
        public const string View = "Permissions.Appointment.View";
        public const string Create = "Permissions.Appointment.Create";
        public const string Edit = "Permissions.Appointment.Edit";
        public const string Delete = "Permissions.Appointment.Delete";
    }

    public static class Verification
    {
        public const string View = "Permissions.Verification.View";
        public const string Process = "Permissions.Verification.Process";
    }

    public static List<string> GetAllPermissions()
    {
        return ReflectionHelper.GetConstants<string>(typeof(PermissionConstants));
    }
}
